v2/robotmk/src/schtasks.rs

```diff
@@ -0,0 +1,265 @@
+use super::command_spec::CommandSpec;
+use super::logging::log_and_return_error;
+use super::session::RunOutcome;
+use super::termination::{kill_process_tree, TerminationFlag};
+
+use anyhow::{bail, Context, Result};
+use async_std::{future::timeout, task::sleep};
+use camino::{Utf8Path, Utf8PathBuf};
+use chrono::{Duration as ChronoDuration, Local};
+use futures::executor;
+use log::{debug, error};
+use std::fs::{read_to_string, write};
+use std::process::Command;
+use std::str::FromStr;
+use std::time::Duration;
+use sysinfo::Pid;
+
+pub fn run_task(task_spec: &TaskSpec) -> Result<RunOutcome> {
```

Comment:
Could we try having explicit errors here?

Reply:
As discussed, this is not the right place to try this out